### PR TITLE
Improve SC docs

### PR DESCRIPTION
**Pull request by velzevur**
Fix slash inconsitencies and add a section regarding what channel state is

### CODE REVIEW

File: channels/ON-CHAIN.md [hunk 1]
```diff
@@ -709,35 +709,24 @@ pushed in subsequent transactions. It is up to participants to decide if they
...
-If the payload is empty, the last on-chain persisted state and `solo_round` are
-used. In this case the proof of inclusion root hash MUST be equal to the one
-persisted for the channel on-chain. If that state was produced unilaterally,
-i.e. via a `channel_force_progress_tx`, making
-`Channel(channel_id).round != Channel(channel_id).solo_round`, the slash is
-based on the `solo_round` and thus can still be disputed.
-
-- `Channel(channel_id).locked_until := Block.height + Channel(channel_id).lock_period`
-
+The payload must be an authenticated by both off-chain state transaction.
```
Comment (hanssv) [inline channels/ON-CHAIN.md]:
```suggestion
The payload must be an off-chain state transaction authenticated by both peers.
```

File: channels/ON-CHAIN.md [hunk 2]
```diff
@@ -709,35 +709,24 @@ pushed in subsequent transactions. It is up to participants to decide if they
...
 
 #### Off-chain transaction payload
 
-If the payload is a transaction it MUST be a `channel_offchain_tx`. It MUST be
-authenticated by both peers.
+The payload is a transaction and it MUST be a `channel_offchain_tx` one. It
```
Comment (hanssv) [inline channels/ON-CHAIN.md]:
```suggestion
The payload is a transaction and it MUST be a `channel_offchain_tx`. It
```

File: channels/ON-CHAIN.md (hunk 1 above)
Comment (hanssv) [reply channels/ON-CHAIN.md]: This is more or less repeated just below, might be up for removal 🤔

File: channels/ON-CHAIN.md [hunk 3]
```diff
@@ -709,35 +709,24 @@ pushed in subsequent transactions. It is up to participants to decide if they
...
 Payload is a valid transaction that has:
 
 * `state_hash` equal to the proof of inclusion's root hash. This is a proof
   that the PoI is correct
 * `channel_id` being the same as the transaction `channel_id`
-* `round` greater than `Channel(channel_id).round`. If
-  `Channel(channel_id).solo_round > Channel(channel_id).round`, then
-  this slash will invalidate progress produced on-chain
+* `round` greater than the last on-chain provided co-authenticated channel
+  state. If the latest on-chain rounds were produced by a
```
Comment (hanssv) [inline channels/ON-CHAIN.md]:
```suggestion
  state. If the latest on-chain round(s) were produced by
```

File: channels/ON-CHAIN.md [hunk 4]
```diff
@@ -709,35 +709,24 @@ pushed in subsequent transactions. It is up to participants to decide if they
...
+  `channel_force_progress_tx` transactions that were based on an older channel
```
Comment (hanssv) [inline channels/ON-CHAIN.md]:
```suggestion
  `channel_force_progress_tx` transaction(s) that were based on an older channel
```
```suggestion
  `channel_force_progress_tx` transactions that were based on an older channel
```

File: channels/state.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+The off-chain state has the exact same structure as the on-chain state. As
+described in [serialization document](/serializations.md#state-trees) the
+off-chain state consists of subtrees. Those are:
+
+* accounts - where accounts and contract balances are stored
+
+* contracts - where the contracts and their states are stored
+
+* calls - where the contract call from the last update are stored. This tree
+  is being purged at every round so the state of the channel is not increased
```
Comment (hanssv) [inline channels/state.md]:
```suggestion
  is being purged at every round so the state of the channel is not enlarged
```

File: channels/state.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  by calls. This is useful especially in long running channels with a lot of
```
Comment (hanssv) [inline channels/state.md]:
```suggestion
  by calls. This is important especially for long running channels with many
```

File: channels/state.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  calls. This is really important for forced progress transaction as if the
```
Comment (hanssv) [inline channels/state.md]:
```suggestion
  calls. This has implications for forced progress transactions as if the
```

File: channels/ON-CHAIN.md (hunk 1 above)
Comment (velzevur) [reply channels/ON-CHAIN.md]: Removed

**Pull request merged**